app/services/auth_token_generator_service.rb

```diff
@@ -0,0 +1,30 @@
+class AuthTokenGeneratorService
+  def self.call(*args)
+    new.call(*args)
+  end
+
+  def call(subscriber, redirect: nil, expiry: 1.week.from_now)
+    data = token_data(subscriber, redirect, expiry)
+    JWT.encode(data, secret, "HS256")
+  end
+
+  private_class_method :new
+
+private
+
+  def token_data(subscriber, redirect, expiry)
+    {
+      "data" => {
+        "subscriber_id" => subscriber.id,
+        "redirect" => redirect,
+      },
+      "exp" => expiry.to_i,
```

Comment:
Do we want to consider validating that `expiry` is in the future? I can't see it being a big problem, and it is low risk given we are only calling it from one place, but we can, by the looks of it, set an expiry that's already past.

Reply:
Yeah it's quite possible, although this is only used internally - similarly a token with a very very short expiry time in the present is pretty pointless. I think it's ok that the onus is on the consumer if they send this service bad data.